File: backend/app/schemas/analytics.py
"""
Pydantic schemas for City Intelligence & Analytics (Phase 8).
"""

from pydantic import BaseModel


class CategoryBreakdownItem(BaseModel):
    category_name: str
    count: int


class DepartmentBreakdownItem(BaseModel):
    department_name: str
    count: int


class PriorityBreakdownItem(BaseModel):
    priority: str
    count: int


class StatusBreakdownItem(BaseModel):
    status: str
    count: int


class AnalyticsSummaryResponse(BaseModel):
    total_complaints: int
    sla_compliance_rate: float
    categories: list[CategoryBreakdownItem]
    departments: list[DepartmentBreakdownItem]
    priorities: list[PriorityBreakdownItem]
    statuses: list[StatusBreakdownItem]


class TrendDataPoint(BaseModel):
    date: str
    count: int


class HotspotClusterItem(BaseModel):
    id: str
    location_name: str
    latitude: float
    longitude: float
    complaint_count: int
    primary_category: str | None = None

File: frontend/src/types/analytics.ts
export interface CategoryBreakdownItem {
  category_name: string;
  count: number;
}

export interface DepartmentBreakdownItem {
  department_name: string;
  count: number;
}

export interface PriorityBreakdownItem {
  priority: string;
  count: number;
}

export interface StatusBreakdownItem {
  status: string;
  count: number;
}

export interface AnalyticsSummaryResponse {
  total_complaints: number;
  sla_compliance_rate: number;
  categories: CategoryBreakdownItem[];
  departments: DepartmentBreakdownItem[];
  priorities: PriorityBreakdownItem[];
  statuses: StatusBreakdownItem[];
}

export interface TrendDataPoint {
  date: string;
  count: number;
}

export interface HotspotClusterItem {
  id: string;
  location_name: string;
  latitude: number;
  longitude: number;
  complaint_count: number;
  primary_category?: string | null;
}
